src/actions/profile-view.js

```diff
@@ -1348,10 +1353,22 @@ export function changeTimelineType(timelineType: TimelineType): Action {
   };
 }
 
-export function changeProfileName(profileName: string | null): Action {
-  return {
-    type: 'CHANGE_PROFILE_NAME',
-    profileName,
+export function changeProfileName(
+  profileName: string | null
+): ThunkAction<Promise<void>> {
+  return async (dispatch, getState) => {
+    if (window.indexedDB) {
+      const hash = getHash(getState());
+      const storedProfile = await retrieveProfileData(hash);
+      if (storedProfile && storedProfile.name !== profileName) {
+        (storedProfile: any).name = profileName;
```

Comment:
Instead of using `any` it's a good idea to work in a "immutable" way. In this case this is as easy as :
```js
const newProfileData = {
  ...storedProfile,
  name: profileName,
};
```

Reply:
Fixed it based on your comments. Thanks